src/game_player.cpp

```diff
@@ -185,9 +185,50 @@ void Game_Player::UpdateScroll(int old_x, int old_y) {
 	}
 }
 
+void Game_Player::UpdateVehicleActions() {
+	if (IsAboard()) {
+		auto* vehicle = GetVehicle();
+		if (vehicle) {
+			vehicle->SyncWithPlayer();
+			vehicle->AnimateAscentDescent();
+		}
+	}
+
+	if (IsStopping()) {
+		if (data()->boarding) {
+			// Boarding completed
+			data()->aboard = true;
+			data()->boarding = false;
+			auto* vehicle = GetVehicle();
+			if (vehicle->IsMoveRouteOverwritten()) {
+				vehicle->CancelMoveRoute();
+			}
+			SetMoveSpeed(vehicle->GetMoveSpeed());
+			vehicle->SetDirection(GetDirection());
+			vehicle->SetSpriteDirection(Left);
+			// Note: RPG_RT ignores the lock_facing flag here!
+			SetSpriteDirection(Left);
+			vehicle->SetX(GetX());
+			vehicle->SetY(GetY());
+		}
+
+		if (data()->unboarding) {
+			// Unboarding completed
+			data()->unboarding = false;
+		}
+	}
+}
+
 void Game_Player::UpdateSelfMovement() {
-	if (!Game_Map::GetInterpreter().IsRunning() && !Game_Map::IsAnyEventStarting()) {
-		if (IsMovable()) {
+	if (!IsBoardingOrUnboarding()
+			&& !Game_Map::GetInterpreter().IsRunning()
+			&& !Game_Message::message_waiting
+			&& !Game_Message::visible
+			&& !IsMoveRouteOverwritten()
+			&& !IsPaused() // RPG_RT compatible logic, but impossible to set pause on player
+			&& !Game_Map::IsAnyEventStarting())
```

Comment:
imo this could be in "IsMovable". When you look into IsMovable almost all of this is already in there (Graphics::TransitionPending is nonsense and can be left out).
Also IsMovable is currently dead code, nothing calls it.
IsMovable even contains a case you forgot:

![bla](https://user-images.githubusercontent.com/1331889/56926303-c35ef480-6ad1-11e9-9b47-eea4cd419fdd.png)

(fancy APNG, should be animated)

Reply:
Nice catch! Thanks